homeassistant/helpers/reload.py

```diff
@@ -77,8 +77,9 @@ async def _resetup_platform(
     if hasattr(component, "async_reset_platform"):
         # If the integration has its own way to reset
         # use this method.
-        await component.async_reset_platform(hass, integration_name)
-        await component.async_setup(hass, root_config)
+        async with hass.data.setdefault("lock_async_reset_platform", asyncio.Lock()):
```

Comment:
What do you think about my question?

Reply:
I don't think that is a problem because the new code guards the component reload that includes the setup. The initial setup seems not giving any issues again.

Reply:
The check at line 120 was previously hit because I didn't reset the dispatcher variable.

Reply:
Ok.